First version Tureng command - Turkish-English dictionary

myucesan opened the pull request:
## Description

<!-- A short summary of your change. If you add a new extension or command, explain what it does. -->
A Raycast command that translates English and Turkish words bidirectionally. Icon by Freepik and API provided by Mehmet Burak Erman.
## Type of change

<!-- Please choose one option and delete others that are not relevant. -->

- New extension

## Screencast
![CleanShot 2021-12-16 at 00 29 06](https://user-images.githubusercontent.com/4138641/146280724-43a5974a-0621-4b17-997f-ef089a654da6.png)

<!-- If you add a new extension or command, include a screencast (or screenshot for a small change). Good screencast will make the review much faster, especially your extension requires registration in other services.  -->

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

pernielsentikaer commented on extensions/tureng/package.json:
```diff
@@ -0,0 +1,36 @@
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "tureng",
+  "title": "Tureng Turkish-English Dictionary",
+  "description": "Translate Turkish words to English and the other way around\n",
```
```suggestion
  "description": "Translate Turkish words to English and the other way around",
```

pernielsentikaer commented on extensions/tureng/package.json:
```diff
@@ -0,0 +1,36 @@
+  "commands": [
+    {
+      "name": "index",
+      "title": "tureng",
+      "subtitle": "Translate English and Turkish words bidirectionally",
```
```suggestion
      "title": "Dictionary",
      "subtitle": "Tureng",
```

pernielsentikaer reviewed (changes requested):
Hi @myucesan 👋 

Thanks for your contribution 💯 Some comments and suggestions for you to work with:

- It would make more sense to let it start on “No Results” screen instead of the placeholder-
- You forgot to add node-fetch to package.json
- Remove debug messages

Is it correct that it only works with Turkish words, inserting "Cat" leads to multiple Cats 😄 

![UrfDMtBW4nPmUDD5](https://user-images.githubusercontent.com/38033107/146325279-7b636830-9758-49f5-9f55-a471dac51825.png)

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

mattisssa commented:
Hey! I've updated the `package.json` to resolve issues with schema and let the workflow build the extension. 

But seems like the extension still contain some errors inside:
>💥 Error: build failed: Could not resolve "node-fetch"

myucesan commented:
> Hi @myucesan 👋
> 
> Thanks for your contribution 💯 Some comments and suggestions for you to work with:
> 
> * It would make more sense to let it start on “No Results” screen instead of the placeholder-
> * You forgot to add node-fetch to package.json
> * Remove debug messages
> 
> Is it correct that it only works with Turkish words, inserting "Cat" leads to multiple Cats 😄
> 
> ![UrfDMtBW4nPmUDD5](https://user-images.githubusercontent.com/38033107/146325279-7b636830-9758-49f5-9f55-a471dac51825.png)
> 
> Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

[x] Shows No results now
[x] added node fetch
[x] removed debug messages
[x] There instead was a mistake with "Cat". All the words are now lowercased as the API, and Tureng handles it correctly like this. Good spot!

Thanks!

pernielsentikaer reviewed (changes requested):
Hi @myucesan 🔥 

We're almost there! Awesome corrections, and the cat is working as expected 😄 

There is a loading spinner when loading the extensions, are you able to remove that one since there should be no loading while loading?

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

myucesan commented:
> Hi @myucesan 🔥
> 
> We're almost there! Awesome corrections, and the cat is working as expected 😄
> 
> There is a loading spinner when loading the extensions, are you able to remove that one since there should be no loading while loading?
> 
> Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

Thanks :) Is the latest commit to your satisfaction? I think that's what you meant, hope I'm correct in thinking that.

pernielsentikaer commented:
Hi @myucesan 🔥 

The loading spinner should be there, it's perfect when you search, however, when loading the extension it should be set default to false – when I loaded the extension, I got 4 seconds “loading spinner” 

Sorry for not being clear enough, also, can you look into the errors the linter are complaining about?

mattisssa commented:
@myucesan Keep in mind that your extension still has a lot of errors and cannot compile:

>src/index.tsx(108,43): error TS2552: Cannot find name 'Json'. Did you mean 'JSON'?
125
src/index.tsx(109,42): error TS2552: Cannot find name 'Json'. Did you mean 'JSON'?
126
src/index.tsx(110,27): error TS7006: Parameter 'jsonResult' implicitly has an 'any' type.
127
src/index.tsx(111,38): error TS2304: Cannot find name 'Json'.

myucesan commented:
> @myucesan Keep in mind that your extension still has a lot of errors and cannot compile:
> 
> > src/index.tsx(108,43): error TS2552: Cannot find name 'Json'. Did you mean 'JSON'?
> > 125
> > src/index.tsx(109,42): error TS2552: Cannot find name 'Json'. Did you mean 'JSON'?
> > 126
> > src/index.tsx(110,27): error TS7006: Parameter 'jsonResult' implicitly has an 'any' type.
> > 127
> > src/index.tsx(111,38): error TS2304: Cannot find name 'Json'.

Yes, this is strange. I'm looking into it.

pernielsentikaer commented:
Hi @myucesan 👋 

What's the status of the PR, are you ready to get it done?

myucesan commented:
> Hi @myucesan 👋 
> 
> 
> 
> What's the status of the PR, are you ready to get it done?

Hi there, 

I am a bit busy atm. Had to go abroad and now exams I'll expect to get to it in February.

Sorry for the wait.

fe9lix commented:
Hey guys, any updates here or should we close and later reopen?

myucesan commented:
> Hi @myucesan 👋
> 
> What's the status of the PR, are you ready to get it done?

I'm done :) Committed.

myucesan commented:
> Hey guys, any updates here or should we close and later reopen?

It's done.

pernielsentikaer commented on extensions/tureng/src/index.tsx:
```diff
@@ -0,0 +1,139 @@
+}
+
+function useSearch() {
+  const [state, setState] = useState<SearchState>({ results: [], isLoading: false });
```
```suggestion
  const [state, setState] = useState<SearchState>({ results: [], isLoading: true });
```
Fixing last part of the [empty state flicker](https://developers.raycast.com/basics/prepare-an-extension-for-store#empty-states)

pernielsentikaer commented on extensions/tureng/src/index.tsx:
```diff
@@ -0,0 +1,139 @@
+import {
+  ActionPanel,
+  CopyToClipboardAction,
```
```suggestion
```
Remove unused import

myucesan commented:
Check last commit, you mean this? It showed a brief loading screen because of it.

myucesan commented:
> @myucesan Could you run prettier on your sources? CI reports some issues: https://github.com/raycast/extensions/runs/5185777181?check_suite_focus=true

done, sorry.

pernielsentikaer commented:
Hi @myucesan 👋 

No, no the `isLoading={state.isLoading}` should indeed be there, but when I open the extension i makes a blank XHR request to `https://tureng.com/en/turkish-english/` with the answer: 

```
{
  about: 'This is Unofficial The Tureng Dictionary JSON API where you can search English & Turkish words by typing /api/:word',
  example: 'https://tureng-api.vercel.app/api/butterfly',
  madeWith: 'node.js and  https://tureng.com',
  createdBy: 'Mehmet Burak Erman'
}
```

This initial call (when searchText.length === 0) should NOT trigger a XHR request but return empty array instead

Does it make sense, otherwise don't hesitate to write me 💪

myucesan commented:
İ'm going to check this, had exams sorry for the delay.

mattisssa commented:
Hey @myucesan! What is the status of this PR? It seems like it's been here for a long time unfinished. Shall we probably close this one and once you're ready – open another PR?
